### ISSUE TITLE
Accessibility: Editor stops announcing file contents to Voice Over after reaching the end of the file

### ISSUE THREAD

**Issue by jquesada2016** [microsoft/vscode#274014]
Type: <b>Bug</b>

1. Open VoiceOver
2. Open a larger file
3. Press `Cmd + down arrow` to go to the end of the file
3. Navigate with the arrows and notice the content that gets announced is not right at all

VS Code version: Code - Insiders 1.106.0-insider (Universal) (00e7d11c2ef29538a0395dab7a5bf4be8719d790, 2025-10-27T20:22:50.391Z)
OS version: Darwin arm64 25.0.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M4 Max (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: enabled_on<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|1, 2, 2|
|Memory (System)|128.00GB (74.07GB free)|
|Process Argv|--crash-reporter-id f861a3fa-a5fd-44d8-a7ee-fb99798a4b7f|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
rust-analyzer|rus|0.4.2659

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
pythonvspyt551cf:31249598
nativeloc1:31118317
dwcopilot:31158714
dwoutputs:31242946
copilot_t_ci:31333650
g012b348:31231168
6gi0g917:31259950
pythonrdcb7:31268811
pythonpcpt1cf:31399617
6518g693:31302842
4f60g487:31327383
6abeh943:31336334
envsactivate1:31349248
editstats-enabled:31346256
aa_c:31379597
asdad-cf:31397862
cloudbuttont:31366566
aihoversummaries_t:31405081
todos-1:31405333
qwen_all_req:31396049
nb255704_tf:31377673
3efgi100_wstrepl:31403338
trigger-command-fix:31379601
chat-sessions-view-t:31407655
auto_model_enabled:31385282
use-responses-api:31390341
applesignin:31399434
f1721337:31405248
afag3459:31405336
control_hh9fb551:31405710
agentic_ask_h7d75428:31406124
ec5jj548:31408805
d7ga2641-codex:31409637

```

</details>

Here's a screen recording of the issue:
https://github.com/user-attachments/assets/0e272fdb-cde1-4a66-93db-c211743f0885

**Comment by meganrogge** [comment]
Did this start happening recently?

**Comment by jquesada2016** [comment]
I noticed the issue started after updating to macOS 26, though this might have been coincidental, but I've been dealing with it for about a month now.

**Comment by meganrogge** [comment]
I can reproduce.

**Comment by vadimkholodilo** [comment]
I'm also seeing a related variant on the current stable build.

- VS Code: `1.115.0`
- Commit: `41dd792b5e652393e7787322889ed5fdc58bd75b`
- OS: macOS `26.3.1`
- Screen reader: VoiceOver

Minimal repro:
1. Open or create a file with multiple lines.
2. Leave the final line empty (trailing newline / blank last line).
3. Put the caret on that empty final line.
4. Press `Up Arrow` to move to the previous line.

Expected:
VoiceOver should announce the full previous line consistently.

Actual:
VoiceOver sometimes announces only a single random symbol/character from the previous line instead of the whole line, and line reading becomes unreliable around the end of the file.

This feels closely related to the end-of-file / wrong-announcement behavior tracked here.
